src/openforms/submissions/form_logic.py

```diff
@@ -86,22 +88,28 @@ def evaluate_form_logic(
     # on that.
     rules = getattr(submission.form, "_cached_logic_rules", None)
     if rules is None:
-        rules = FormLogic.objects.filter(form=submission.form)
+        # TODO the ordering on PK will be replaced by ordering on an order field once it's implemented
+        rules = FormLogic.objects.filter(form=submission.form).order_by("pk")
         submission.form._cached_logic_rules = rules
 
     submission_state = submission.load_execution_state()
 
-    updated_data = deepcopy(step.data)
+    updated_step_data = deepcopy(step.data)
+    updated_submission_data = deepcopy(data)
     for rule in rules:
-        if jsonLogic(rule.json_logic_trigger, data):
+        if jsonLogic(rule.json_logic_trigger, updated_submission_data):
             for action in rule.actions:
                 action_details = action["action"]
+                # TODO this action will be replaced by changing the value of variables
                 if action_details["type"] == LogicActionTypes.value:
-                    new_value = jsonLogic(action_details["value"], data)
+                    new_value = jsonLogic(
+                        action_details["value"], updated_submission_data
+                    )
                     configuration = set_property_value(
                         configuration, action["component"], "value", new_value
                     )
```

Comment:
Formio itself keeps the configuration and values separated, so `value` inside of the component configuration is ignored by formio. The only reason it would be there, is that we use it for some reason.

I was curious because I saw some code duplication between `value` and `variable` action types, but I agree that we should probably just leave it and then remove it all once variables is done.